Add cmesh example triangulated spherical surface [1/n]

jmark opened the pull request:
**_Describe your changes here:_**

Title says it all. This is a follow-PR to https://github.com/DLR-AMR/t8code/pull/753.

In addition to that, I extended the vtk output by a `diameter` field which is reveals how large the difference
between element sizes is. Generally, it is the goal to have elements of equal sizes in case of uniform refinement.
![grafik](https://github.com/DLR-AMR/t8code/assets/10619309/eb7937c6-fe1c-4c61-b0cc-d5d52cb01dc6)

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.


#### General
- [x] The reviewer executed the new code features at least once and checked the results manually

- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] New source/header files are properly added to the Makefiles
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [x] The code is covered in an existing or new test case using Google Test

#### Github action

- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [x] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [x] The author added a BSD statement to `doc/` (or already has one)

Davknapp commented on src/t8_cmesh/t8_cmesh_examples.c:
```diff
@@ -2738,3 +2738,73 @@ t8_cmesh_new_squared_disk (const double radius, sc_MPI_Comm comm)
+    all_eclasses[itree] = T8_ECLASS_TRIANGLE;
+  }
+
+  const double r = radius; /* Only for readability. */
```
```suggestion
```
We should use explaining variable names. radius is actually a good name.

Davknapp commented on src/t8_cmesh/t8_cmesh_examples.c:
```diff
@@ -2738,3 +2738,73 @@ t8_cmesh_new_squared_disk (const double radius, sc_MPI_Comm comm)
+
+  const double r = radius; /* Only for readability. */
+
+  double vertices_top[3][3] = { { r, 0.0, 0.0 }, { 0.0, r, 0.0 }, { 0.0, 0.0, r } };
```
```suggestion
  double vertices_top[3][3] = { { radius, 0.0, 0.0 }, { 0.0, radius, 0.0 }, { 0.0, 0.0, radius } };
```

Davknapp commented on src/t8_cmesh/t8_cmesh_examples.c:
```diff
@@ -2738,3 +2738,73 @@ t8_cmesh_new_squared_disk (const double radius, sc_MPI_Comm comm)
+
+  double vertices_top[3][3] = { { r, 0.0, 0.0 }, { 0.0, r, 0.0 }, { 0.0, 0.0, r } };
+
+  double vertices_bot[3][3] = { { r, 0.0, 0.0 }, { 0.0, r, 0.0 }, { 0.0, 0.0, -r } };
```
```suggestion
  double vertices_bot[3][3] = { { radius, 0.0, 0.0 }, { 0.0, radius, 0.0 }, { 0.0, 0.0, -radius } };
```

Davknapp commented on src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx:
```diff
@@ -113,6 +113,224 @@ t8_geometry_squared_disk::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreei
+  t8_locidx_t ltreeid = t8_cmesh_get_local_id (cmesh, gtreeid);
+  double *tree_vertices = t8_cmesh_get_tree_vertices (cmesh, ltreeid);
+
+  n[0] = tree_vertices[0] + tree_vertices[3] + tree_vertices[6];
```
We have a functions for such vector-operations. Please use them.

jmark replied:
Done.

Davknapp commented on src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx:
```diff
@@ -113,6 +113,224 @@ t8_geometry_squared_disk::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreei
+                                                              const double *ref_coords, const size_t num_coords,
+                                                              double *out_coords) const
+{
+  double n[3]; /* Normal vector of the current triangle. */
```
Please use more explaining names, like normal, radial, position for example.

jmark replied:
Done.

Davknapp reviewed (changes requested):
Loogs good in general, please address my comments.

Davknapp commented on src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx:
```diff
@@ -113,6 +113,224 @@ t8_geometry_squared_disk::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreei
+  }
+
+  /* Init output coordinates with zeros. */
+  for (size_t i_coord = 0; i_coord < num_coords; i_coord++) {
```
I don't think you need to zero-initialize, as in line 221 - 223 you can directly set the out_coords to R*p[0]

jmark replied:
Done.

jmark commented:
@Davknapp I refactored the code. I realized that I did some unnecessary computations which I removed in this version. The code also becomes simpler that way.
